http/typescript/fetch/src/httpClient.ts

```diff
@@ -17,17 +17,17 @@ export class HttpClient {
 	 * @param {(request: string, init?: RequestInit) => Promise < Response >} custom fetch function - a Fetch API implementation
 	 *
 	 */
-	public constructor(private customFetch?: (request: string, init?: RequestInit) => Promise<Response>, ...middlewares: Middleware[]) {
+	public constructor(private customFetch?: (request: string, init: RequestInit) => Promise<Response>, ...middlewares: Middleware[]) {
 		// Use default middleware chain if middlewares and custom fetch function are  undefined
 		if (!middlewares.length || middlewares[0] === null) {
-			if (this.customFetch) {
+			if (this.customFetch !== undefined) {
 				this.setMiddleware(...MiddlewareFactory.getDefaultMiddlewareChain(customFetch));
```

Comment:
we should not need 4 cases for that, but only 3 (middleware handlers provided or not), and let the get default middleware method handle the optional defaulted parameter.

Reply:
Done